tests/sip/test_client.py

```diff
@@ -14,17 +14,29 @@
 
 class MockSocket(object):
     def __init__(self, *args, **kwargs):
+        self.data = b''
         self.args = args
         self.kwargs = kwargs
         self.timeout = None
         self.connected_to = None
 
+    def queue_data(self, new_data):
+        if isinstance(new_data, str):
+            new_data = new_data.encode("cp850")
+        self.data += new_data
+
     def connect(self, server_and_port):
         self.connected_to = server_and_port
 
     def settimeout(self, value):
         self.timeout = value
 
+    def recv(self, size):
+        block = self.data[:size]
+        self.data = self.data[size:]
+        return block
+
+
```

Comment:
This makes `MockSocket` more like a real socket, allowing us to add test coverage to `read_message`.